signer/src/api/mod.rs

```diff
@@ -16,3 +22,10 @@ pub struct ApiState<C> {
 
 /// The name of the sbtc registry smart contract.
 const SBTC_REGISTRY_CONTRACT_NAME: &str = "sbtc-registry";
+
+enum UpdateResult {
+    Deposit(Result<UpdateDepositsResponse, Error>),
+    Withdrawal(Result<UpdateWithdrawalsResponse, Error>),
+    CreatedWithdrawal(Vec<Result<Withdrawal, Error>>),
+    Chainstate(Result<Chainstate, Error>),
+}
```

Comment:
nit: We should probably just move this to where it is used. We've been trying to keep our `mod.rs` files pretty sparse.

Reply:
Got it!